### DISCUSSION TITLE
Questions Regarding the Development of an Explicit Dynamics Kernel

### DISCUSSION THREAD

**Discussion by Yasunaoka** [idaholab/moose#25792] [Q&A Modules: Solid mechanics] [answered]
Dear Community

I am currently utilizing the DynamicMaster, and particularly interested in explicit dynamics which is suitable for the high strain rate problem(about 1e4~1e6 s^-1). However, I have checked the source code of DynamicMaster, and found that this kernel only supports a handcoded implicit hht format.

I have two questions, and I would greatly appreciate your assistance and guidance:

1. Firstly, I would like to inquire if there are any plans by the official team to introduce an explicit dynamics module to the Tensor Mechanical? If such plans exist, could you provide insights into the expected timeline or milestones for this development?

2. If I intend to code an explicit dynamics kernel myself, could you offer any advice or guidance on where to begin? Such guidance would be invaluable for my project within Moose.In addition, the explicit dynamics solution format itself isn't complicated. However, I observed that the Jacobi matrix is always computed in the MOOSE solution procedure, but the Jacobi matrix is not required in explicit dynamics and there is no convergence problem. Does this feature conflict with and MOOSE's solution framework?

Thank you very much for your time and assistance, and I look forward to your response.

Sen

**Comment by maxnezdyur** [comment]
@recuero @dschwen IIRC, there is work that will be starting soon about explicit dynamics, but I can't comment on the timeline.

**Reply by recuero** [reply]
It's not clear to me that an explicit dynamics module will be added. That work can be done within Tensor Mechanics. You can already perform solid mechanics simulations with a central difference scheme (look at MOOSE and MASTODON input files).

We plan on doing development in that area within the next couple of years. But I don't know exactly where your interests are, so I can't provide more specifics.

Regarding your last question. The fact that MOOSE typically builds a Jacobian matrix isn't a big hurdle to do explicit. As I mentioned above, that's already possible with the code in the repository.

**Reply by Yasunaoka** [reply]
Hi recuero, firstly thank you for your kind help. The problem I am interested in is the fragmentation of a metallic cylindrical shell under explosive loading, where the material will be subjected to tens of Gpa of loading in a few microseconds - an extreme nonlinear problem involving fracture. If computed using implicit integration, the extreme nonlinearity will lead to difficulties in convergence ( which I have already encountered) as the stiffness matrix needs to be computed and Jacobi matrices introduced at each step. In contrast, in the explicit dynamics computational format, the information on the current time can be obtained directly from the information on the previous step, without the need to introduce Jacobi matrices and iterations, which is very favourable for solving impact fracture problems. This is why I wanted to compute this problem via explicit dynamics.
However, in the MOOSE code, the solution format in DynamicStressDivergenceTensors is a hard-coded form of Hilber-Hughes-Taylor time integration - which is an implicit format. Therefore, I don't think that the central difference scheme in executioner overrides this solution format, and I think it may only be in effect when calculating the displacement variable, refer to this discussion: #24896.
Therefore, I would still like to ask this question: does this feature of showing that no Jacobi matrix is needed in the dynamics computation conflict with moose's solution framework? Many thanks!

**Reply by recuero** [reply, marked as answer]
> does this feature of showing that no Jacobi matrix is needed in the dynamics computation conflict with moose's solution framework? 

There is no conflict with the solution framework. You should be getting an explicit solution scheme when you set up the problem as explicit. 

> which is an implicit format

There exists the opportunity to mix HHT and central difference given the input parameters of the dynamic tensor mechanics action, but that's not a good idea.

**Reply by Yasunaoka** [reply]
Hi recuero, thanks for your patience, the explicit solver is now running successfully.

**Reply by recuero** [reply]
Great!
